src/libraries/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -639,6 +639,32 @@ public void AssemblyResolve_IsNotCalledForCoreLibResources()
             }).Dispose();
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/43909", TestRuntimes.Mono)]
+        public void AssemblyResolve_ExceptionPropagatesCorrectly()
+        {
+            RemoteExecutor.Invoke(() => {
+                bool handlerExceptionThrown = false;
+
+                AppDomain.CurrentDomain.AssemblyResolve += (sender, e) =>
+                {
+                    throw new Exception("Failure");
+                };
+
+                try
+                {
+                    Type.GetType("Some.Assembly.Type, Some.Assembly");
```

Comment:
maybe make this a `[Theory]` and test both `throwOnError` `true` and `false` cases (they behave the same on CoreCLR)